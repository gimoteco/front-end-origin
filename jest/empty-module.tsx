import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
