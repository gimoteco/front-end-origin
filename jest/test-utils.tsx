import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
