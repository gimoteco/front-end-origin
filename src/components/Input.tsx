import * as React from 'react';
import MoneyInput from './MoneyInput';
import MonthInput from './MonthInput';

export enum InputType {
  Money,
  Month
}

export type InputValueType = string | number | Date;

interface InputProps {
  id: string;
  type: InputType;
  value: InputValueType;
  onChange: (newValue: InputValueType) => void;
}

function Input({ id, type, value, onChange }: InputProps) {
  if (type === InputType.Money)
    return <MoneyInput value={value as number} id={id} onChange={onChange} />;
  else if (type === InputType.Month)
    return <MonthInput id={id} value={value as Date} onChange={onChange} />;
  return null;
}

export default React.memo(Input);
