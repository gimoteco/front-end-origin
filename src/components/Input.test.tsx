import React from 'react';
import { render } from '../../jest/test-utils';
import Input, { InputType } from './Input';

describe('Input', () => {
  it('should render money input according snapshot', () => {
    const { container } = render(
      <Input
        id="some-id"
        onChange={jest.fn()}
        value={40000}
        type={InputType.Money}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render month input according snapshot', () => {
    const { container } = render(
      <Input
        id="some-id"
        onChange={jest.fn()}
        value={new Date(2200, 0, 1)}
        type={InputType.Month}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should not render unknown input type', () => {
    const typeNotImplemented = 9999;
    const { container } = render(
      <Input
        id="some-id"
        onChange={jest.fn()}
        value={new Date(2200, 0, 1)}
        type={typeNotImplemented}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
