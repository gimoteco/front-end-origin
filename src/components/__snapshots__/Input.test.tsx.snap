// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should not render unknown input type 1`] = `<div />`;

exports[`Input should render money input according snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa djcBRL"
  >
    <label
      class="sc-htpNat eRwJqC"
      for="some-id"
    >
      $
    </label>
    <input
      class="sc-bwzfXH cvgptw"
      data-testid="some-id"
      id="some-id"
      type="text"
      value="40000"
    />
  </div>
</div>
`;

exports[`Input should render month input according snapshot 1`] = `
<div>
  <div
    class="sc-ifAKCX cNXLUN"
    data-testid="some-id"
    id="some-id"
    tabindex="0"
  >
    <button
      class="sc-bxivhb dsqJjx"
      type="button"
    >
      <span />
    </button>
    <div
      class="sc-EHOje jAvrM"
    >
      <span
        class="sc-bZQynM hkRXLa"
      >
        January
      </span>
      <span
        class="sc-gzVnrw ioEuae"
      >
        2200
      </span>
    </div>
    <button
      class="sc-bxivhb dsqJjx"
      type="button"
    >
      <span />
    </button>
  </div>
</div>
`;
